### PR TITLE
Add Libevent package.

### PR DESCRIPTION
**Pull request by isaachier**

### CODE REVIEW

File: cmake/projects/Libevent/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME Libevent
+    VERSION "2.1.8"
+    URL "https://github.com/isaachier/libevent/archive/hunter-release-2.1.8-alpha.tar.gz"
+    SHA1 "fc0e6257bb5798d80969f67c4d346915ebae094d")
+
+hunter_pick_scheme(DEFAULT url_sha1_cmake)
+#hunter_cacheable(Libevent)
```
Comment (ruslo) [inline cmake/projects/Libevent/hunter.cmake]: Problems with `*Config.cmake`?

File: docs/packages/pkg/Libevent.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+.. _pkg.Libevent:
+
+Libevent
+========
+
+-  `Official <https://github.com/libevent/libevent>`__
+-  `Hunterized <https://github.com/hunter-packages/libevent>`__
+-  `Example <https://github.com/ruslo/hunter/blob/master/examples/Libevent/CMakeLists.txt>`__
+-  Added by `Isaac Hier <https://github.com/isaachier>`__ (`pr-N <https://github.com/ruslo/hunter/pull/N>`__)
```
Comment (ruslo) [inline docs/packages/pkg/Libevent.rst]: `N` -> `1019`

File: examples/Libevent/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+include("../common.cmake")
+
+project(download-libevent)
+
+hunter_add_package(Libevent)
+
+find_package(Libevent CONFIG REQUIRED)
+
+add_executable(main main.cpp)
+target_link_libraries(main event)
```
Comment (ruslo) [inline examples/Libevent/CMakeLists.txt]: Doesn't fit the `docs/packages/pkg/Libevent.rst` snippet. Should be `Libevent::even`?

File: examples/Libevent/main.cpp [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#include <event2/event.h>
```
Comment (ruslo) [inline examples/Libevent/main.cpp]: Got error:
```
main.cpp:1:26: fatal error: event2/event.h: No such file or directory
```

File: cmake/projects/Libevent/hunter.cmake (hunk 1 above)
Comment (isaachier) [reply cmake/projects/Libevent/hunter.cmake]: Yes... already foresee issues with this package.

File: examples/Libevent/CMakeLists.txt (hunk 3 above)
Comment (isaachier) [reply examples/Libevent/CMakeLists.txt]: I wish that worked. I think I have to rewrite parts of the Libevent CMake files to fix the install.

File: examples/Libevent/main.cpp (hunk 4 above)
Comment (isaachier) [reply examples/Libevent/main.cpp]: Weird. I thought this part worked. Then again, may be local machine setup. Will try again and update.

File: cmake/projects/Libevent/hunter.cmake (hunk 1 above)
Comment (ruslo) [reply cmake/projects/Libevent/hunter.cmake]: We can do the same as we did with "double-conversion": I can fork project and review fix for install instructions.

File: examples/Libevent/main.cpp (hunk 4 above)
Comment (ruslo) [reply examples/Libevent/main.cpp]: If we are talking about Linux you can use Docker to reproduce issues on Travis CI, see:
* https://github.com/hunter-packages/travis-linux-docker

Comment (isaachier) [reply examples/Libevent/main.cpp]: Good idea. Will do.

File: docs/packages/pkg/Libevent.rst (hunk 2 above)
Comment (isaachier) [reply docs/packages/pkg/Libevent.rst]: Ya meant to do this one I got a pull request number.

File: cmake/projects/Libevent/hunter.cmake (hunk 1 above)
Comment (isaachier) [reply cmake/projects/Libevent/hunter.cmake]: Perfect thanks.

Comment (ruslo) [reply cmake/projects/Libevent/hunter.cmake]: Fork created:
* https://github.com/hunter-packages/libevent

Branch `hunter-2.1.8` from latest upstream release `2.1.8`.

File: docs/packages/pkg/Libevent.rst (hunk 2 above)
Comment (ruslo) [reply docs/packages/pkg/Libevent.rst]: You have it :) It's `1019` - pull request that is currently opened.

File: cmake/projects/Libevent/hunter.cmake [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+include(hunter_add_version)
+include(hunter_cacheable)
+include(hunter_cmake_args)
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME Libevent
+    VERSION "2.1.8"
+    URL "https://github.com/isaachier/libevent/archive/hunter-2.1.8-p5.tar.gz"
```
Comment (ruslo) [inline cmake/projects/Libevent/hunter.cmake]: Please set this release:
* https://github.com/hunter-packages/libevent/releases/tag/v2.1.8-p0

File: cmake/configs/default.cmake [hunk 6]
```diff
@@ -163,6 +164,7 @@ hunter_config(dbus VERSION 1.10.0-hunter-4)
 hunter_config(dest VERSION 0.8.0-p4)
 hunter_config(dlib VERSION 19.6-p0)
 hunter_config(doctest VERSION 1.2.0)
+hunter_config(double-conversion VERSION 3.0.0)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: Doesn't belong here.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2366
* https://travis-ci.org/ingenue/hunter/builds/275019751?utm_source=github_status&utm_medium=notification

**Comment by isaachier** [comment]
Seems like mobile platforms could not find OpenSSL package.

**Comment by ruslo** [comment]
CMake can't use `try_run` while cross-compiling:
```
CMake Error: TRY_RUN() invoked in cross-compiling mode, please set the following cache variables appropriately:

   EVENT__HAVE_WAITPID_WITH_WNOWAIT_EXITCODE (advanced)
```
Workaround for this can be explicitly setting `try_run` output variables, like this:
* https://github.com/hunter-packages/jpeg/blob/17c8acbd717fb9134f0f14e014bee4923d59826b/CMakeLists.txt#L70-L74

File: cmake/projects/Libevent/hunter.cmake [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+include(hunter_add_version)
+include(hunter_cacheable)
+include(hunter_cmake_args)
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME Libevent
+    VERSION "2.1.8"
+    URL "https://github.com/hunter-packages/libevent/archive/v2.1.8-p0.tar.gz"
```
Comment (ruslo) [inline cmake/projects/Libevent/hunter.cmake]: Use `2.1.8-p1` release:
* https://github.com/hunter-packages/libevent/releases/tag/v2.1.8-p1

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2421
* https://travis-ci.org/ingenue/hunter/builds/276923018?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
OSX builds failing because somewhere inside CMakeLists.txt variable with name `0` set to value `inline` :) Meaning condition `if(A EQUAL 0)` now is equivalent to `if(A EQUAL inline)`. Problem similar to:
* http://cgold.readthedocs.io/en/latest/tutorials/control-structures.html#cmp0054

I can introduce protection with this improvement:
* https://github.com/ruslo/hunter/issues/558

For now as a workaround please remove code in libevent that sets `0` to `inline`.

I've added documentation about testing packages on CI servers:
* https://docs.hunter.sh/en/latest/creating-new/create/cmake.html#ci-testing

Please test it before sending new commits.

**Comment by isaachier** [comment]
Got it! https://github.com/hunter-packages/libevent/pull/3

**Comment by isaachier** [comment]
OK just updated to `p2`.

**Comment by ruslo** [comment]
> I've added documentation about testing packages on CI servers

> OK just updated to p2

Have you tested it?

**Comment by isaachier** [comment]
Oh right. I just set up Travis for my fork yesterday, but still not sure how to use the testing table on my branch. I'll try merging it and building on temporary branch.

**Comment by ruslo** [comment]
> not sure how to use the testing table on my branch

Docs are here:
* https://docs.hunter.sh/en/latest/creating-new/create/cmake.html#branches-structure

**Comment by isaachier** [comment]
Awaiting macOS builds: https://travis-ci.org/isaachier/hunter/builds/277907067.

**Comment by isaachier** [comment]
Should be ready to accept.

**Comment by ruslo** [comment]
> Should be ready to accept

Please send CI configs first:
* https://docs.hunter.sh/en/latest/creating-new/create/cmake.html#pull-requests-tests

**Comment by isaachier** [comment]
Sorry forgot to do that. Did it now: https://github.com/ingenue/hunter/pull/134.

**Comment by ruslo** [comment]
Testing:
* https://travis-ci.org/ingenue/hunter/builds/279583790?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2458
* https://travis-ci.org/ingenue/hunter/builds/279604388?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.19.110

**Pull request merged**